src/native/libs/System.Security.Cryptography.Native.Android/pal_cipher.c

```diff
@@ -299,6 +299,57 @@ int32_t AndroidCryptoNative_CipherFinalEx(CipherCtx* ctx, uint8_t* outm, int32_t
     return CheckJNIExceptions(env) ? FAIL : SUCCESS;
 }
 
+
+int32_t AndroidCryptoNative_AeadCipherFinalEx(CipherCtx* ctx, uint8_t* outm, int32_t* outl, int32_t* authTagMismatch)
+{
+    if (!ctx)
+        return FAIL;
+
+    abort_if_invalid_pointer_argument(outm);
+    abort_if_invalid_pointer_argument(outl);
+    abort_if_invalid_pointer_argument(authTagMismatch);
+
+    JNIEnv* env = GetJNIEnv();
+
+    *outl = 0;
+    *authTagMismatch = 0;
+
+    jbyteArray outBytes = (jbyteArray)(*env)->CallObjectMethod(env, ctx->cipher, g_cipherDoFinalMethod);
+    jthrowable ex = NULL;
+
+    if (TryGetJNIException(env, &ex, false))
+    {
+        if (ex == NULL)
+        {
+            return FAIL;
+        }
+
+        jclass aeadException = (*env)->FindClass(env, "javax/crypto/spec/AEADBadTagException");
```

Comment:
It seemed wiser at this point to fetch the exception class from the environment instead of caching an instance of it like we do here:

https://github.com/dotnet/runtime/blob/cebac827ff2563b9305a17abe02556eef92eb2cf/src/native/libs/System.Security.Cryptography.Native.Android/pal_jni.c#L694

Caching it create a gref that is in memory forever. I don't anticipate we will need a cached instance of the jclass available, and it seemed like a better tradeoff to fetch the class in the error path of AEAD decryption instead of during Android startup.

Happy to revisit and discuss though.

Reply:
I think we'd rather fetch it once at startup so that we're matching the pattern. A single additional gref isn't an issue given how many we already have.

Reply:
Makes sense. Addressed.